libecole/src/observation/strongbranchingscores.cpp

```diff
@@ -68,12 +68,12 @@ std::optional<xt::xtensor<double, 1>> StrongBranchingScores::extract(scip::Model
 	model.set_param("branching/vanillafullstrong/idempotent", idempotent);
 
 	/* Store strong branching scores in tensor */
-	auto const num_lp_columns = static_cast<std::size_t>(SCIPgetNLPCols(scip));
-	auto strong_branching_scores = xt::xtensor<double, 1>({num_lp_columns}, std::nan(""));
+	auto const nb_vars = static_cast<std::size_t>(SCIPgetNVars(scip));
+	auto strong_branching_scores = xt::xtensor<double, 1>({nb_vars}, std::nan(""));
 
 	for (auto const [var, score] : views::zip(cands, cands_scores)) {
-		auto const lp_index = static_cast<std::size_t>(SCIPcolGetLPPos(SCIPvarGetCol(var)));
-		strong_branching_scores[lp_index] = static_cast<double>(score);
+		auto const var_index = static_cast<std::size_t>(SCIPvarGetProbindex(var));
```

Comment:
This is a major change. For example, the branching tutorial will break. I suggest to make a separate PR for that.

Reply:
Why would the branching tutorial break?

Reply:
The tutorial assumes the scores array (SB or PseudoCost) is ordered according the LP index of the variables. Not their problem index. `scores[action_set]` does return the scores of the branching cancidates.
```
# We will solve problems (run episodes) until we have saved enough samples
max_samples_reached = False
while not max_samples_reached:
    episode_counter += 1
    
    observation, action_set, _, done, _ = env.reset(next(instances))
    while not done:
        (scores, scores_are_expert), node_observation = observation
        action = action_set[scores[action_set].argmax()]
```

Reply:
I'm still not understanding you very well, but I checked and the action set returned by the Branching environment is indices w.r.t. column index
https://github.com/ds4dm/ecole/blob/7100c77a64389ab24e136e3cf0fec30850c88c4d/libecole/src/dynamics/branching.cpp#L17-L28
which is perhaps what you were trying to say? In any case that would be enough to break the tutorial, you are right.

This section has been moved to a separate PR.

Reply:
Yes that's what I meant, the tutorial code assumes column indices, so if we move to problem variable indices we'll have to fix the tutorial as well.